src/main/java/com/dungzi/backend/domain/room/application/RoomService.java

```diff
@@ -0,0 +1,155 @@
+package com.dungzi.backend.domain.room.application;
+
+import com.dungzi.backend.domain.room.dao.RoomAddressDao;
+import com.dungzi.backend.domain.room.dao.RoomDao;
+import com.dungzi.backend.domain.room.dao.RoomInfoDao;
+import com.dungzi.backend.domain.room.dao.RoomOptionDao;
+import com.dungzi.backend.domain.room.domain.Room;
+import com.dungzi.backend.domain.room.domain.RoomAddress;
+import com.dungzi.backend.domain.room.domain.RoomInfo;
+import com.dungzi.backend.domain.room.domain.RoomOption;
+import com.dungzi.backend.domain.room.dto.*;
+import com.dungzi.backend.domain.user.application.AuthService;
+import com.dungzi.backend.domain.user.domain.User;
+import com.dungzi.backend.global.s3.FileUploadService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.*;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class RoomService {
+    private AuthService authService;
+    private final User user = authService.getUserFromSecurity();;
+    private final RoomDao roomDao;
+    private final RoomAddressDao roomAddressDao;
+    private final RoomInfoDao roomInfoDao;
+    private final RoomOptionDao roomOptionDao;
+
+    @Transactional
+    public String saveAction(RoomRequestDto requestDto, List<MultipartFile> files) throws Exception {
+
+        RoomDto roomDto = RoomDto.builder()
+                .userId(user.getUserId().toString())
+                .title(requestDto.getTitle())
+                .content(requestDto.getContent())
+                .image(files.size())        //TO-DO : 날짜는 어떤 형식으로 입력할 지
+                .heartNum(0)
+                .build();
+
+        Room room = new Room();
+        room = roomSave(roomDto);
+        String roomId = (room.getRoomId()).toString();
+
+        FileUploadService.uploadRoomFile(roomId, files);
+
+        requestDto.setRoomId(roomId);
+        roomAddressSave(requestDto);
+        roomInfoSave(requestDto);
+        roomOptionSave(requestDto);
+
+
+        return roomId;
+    }
+
+
+    @Transactional
+    public Room roomSave(RoomDto requestDto) {
+        //requestDto.setUserId(user.getUserId());
+        return roomSave(requestDto.toEntity(requestDto));
+    }
```

Comment:
앗 이미 위에서 @transactional해서 따로 할 필요가 없었네요!
그런데 s3는 트랜잭션 기능이 없어서 파일을 업로드하고 만약 오류가 생길 경우 파일을 다시 삭제하는 걸로 하면될까요?

Reply:
음 그부분이 애매하겠네요... 실제 Transactional 코드내에서 다른 에러(업로드할때 에러나는경우)가 터져도 롤백이 되는지 잘 모르겠습니다...

Reply:
~~~
Yes, if the method annotated with @Transactional throws an exception, even if it's not related to database operations (DAO), the transaction will be rolled back. This is because the purpose of the @Transactional annotation is to demarcate a block of code as a single transaction, and if any exception is thrown within this block, the transaction will be automatically rolled back.
~~~

chatgtp에서는 dao 관련된 로직이 아니고 다른 에러가 터져도 롤백이 된다고 하네요 그대로 둬도 될 거 같아요!

Reply:
넵 감사합니다!